test/functional/Java15andUp/src/org/openj9/test/regularclassesandinterfaces/RegularClassAndInterfaceFinalFieldTests.java

```diff
@@ -0,0 +1,116 @@
+package org.openj9.test.regularclassesandinterfaces;
+
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+import jdk.internal.misc.Unsafe;
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.lang.reflect.Field;
+
+@Test(groups = { "level.sanity" })
+public class RegularClassAndInterfaceFinalFieldTests {
+    static Unsafe unsafe = Unsafe.getUnsafe();
+
+    private class TestClass {
+        static String modifiableField = "old";
+        static final String finalField = "old";
+    }
+
+    private interface TestInterface {
+        String finalField = "old";
```

Comment:
maybe. explicitly add the `static final` keyword to make this more clear (rather than auto done by javac)